tests/test_asset.py

```diff
@@ -25,6 +25,22 @@ def test_init(asset_mock):
     assert asset_mock.asset_id == ASSET_ID
 
 
+def test_init_asset_failure(auth_mock):
```

Comment:
Let's decompose this into 2 simple separate scenarios:
1. test_init_should_fail_if_both_id_and_info_given
2. test_init_should_fail_if_neither_id_nor_info_given